### PR TITLE
Implemented path ends stretching.

### PR DESCRIPTION
**Pull request by ygorshenin**
@mgsergio PTAL

### CODE REVIEW

File: openlr/openlr_simple_decoder.cpp [hunk 1]
```diff
@@ -466,12 +466,39 @@ class AStarRouter
...
+    bool m_isSpecial = false;
+  };
+
   using Links = map<Vertex, pair<Vertex, Edge>>;
 
   class Score
   {
   public:
-    static const int kFakeCoeff = 3;
+    static const int kTrueFakeCoeff = 3;
```
Comment (mgsergio) [inline openlr/openlr_simple_decoder.cpp:501]: Please add a comment to this and the next one.

File: openlr/openlr_simple_decoder.cpp [hunk 2]
```diff
@@ -605,6 +641,139 @@ class AStarRouter
...
+  // Finds the longest prefix of [b, e) that covers edge (u, v).
+  // Returns the fraction of the coverage to the length of the (u, v).
+  template <typename It>
+  double GetMatchingScore(m2::PointD const & u, m2::PointD const & v, It b, It e)
+  {
+    double const kEps = 1e-5;
+
+    double const d = MercatorBounds::DistanceOnEarth(u, v);
+
+    double n = 0;
```
Comment (mgsergio) [inline openlr/openlr_simple_decoder.cpp]: Please rename 'n' to something meaningful.

Comment (mgsergio) [reply openlr/openlr_simple_decoder.cpp]: 'l' for example :)

Comment (ygorshenin) [reply openlr/openlr_simple_decoder.cpp]: n means "numerator". d means "denominator". Quite obvious.

File: openlr/openlr_simple_decoder.cpp [hunk 3]
```diff
@@ -605,6 +641,139 @@ class AStarRouter
...
+      auto const & t = b->second;
+      if (!m2::IsPointOnSegmentEps(s, u, v, kEps) || !m2::IsPointOnSegmentEps(t, u, v, kEps))
+        break;
+      n += MercatorBounds::DistanceOnEarth(s, t);
+    }
+
+    return d == 0 ? 0 : my::clamp(n / d, 0, 1);
+  }
+
+  // Finds the longest prefix of fake edges of [b, e) that matches to
```
Comment (mgsergio) [inline openlr/openlr_simple_decoder.cpp]: s/that matches to/that have the same stage as |stage|

Comment (ygorshenin) [reply openlr/openlr_simple_decoder.cpp]: Done

**Comment by ygorshenin** [comment]
PPP-TAL

File: openlr/openlr_simple_decoder.cpp (hunk 2 above)
Comment (mgsergio) [reply openlr/openlr_simple_decoder.cpp]: I thought 'd' stood for distance and the right part gives a huge hint for it.

**Comment by mgsergio** [comment]
LGTM

**Pull request merged**